mesonbuild/backend/backends.py

```diff
@@ -959,10 +959,10 @@ def generate_basic_compiler_args(self, target: build.BuildTarget, compiler: 'Com
         # Add compile args added using add_global_arguments()
         # These override per-project arguments
         commands += self.build.get_global_args(compiler, target.for_machine)
-        # Using both /ZI and /Zi at the same times produces a compiler warning.
-        # We do not add /ZI by default. If it is being used it is because the user has explicitly enabled it.
-        # /ZI needs to be removed in that case to avoid cl's warning to that effect (D9025 : overriding '/ZI' with '/Zi')
-        if ('/ZI' in commands) and ('/Zi' in commands):
+        # Using both /Z7 or /ZI and /Zi at the same times produces a compiler warning.
+        # We do not add /Z7 or /ZI by default. If it is being used it is because the user has explicitly enabled it.
+        # /Zi needs to be removed in that case to avoid cl's warning to that effect (D9025 : overriding '/Zi' with '/ZI')
+        if (('/ZI' in commands) or ('/Z7' in commands)) and ('/Zi' in commands):
```

Comment:
I'd personally check /Zi first, given that the other part is a tad more expensive, but maybe that's my micro-optimizing database programmer side talking...

Reply:
I have fixed the order of comparisons.

For me this works fine, when I remove my changes, I get the compiler warnings. With my changes these warnings are quiet. Are you sure you are using the changed version of meson and did you build from scratch?

Reply:
> For me this works fine, when I remove my changes, I get the compiler warnings. With my changes these warnings are quiet. Are you sure you are using the changed version of meson and did you build from scratch?

Yes. The reason it doesn't work is that Z7 is only added in the subsequent line, via `self.environment.coredata.get_external_args(target.for_machine, compiler.get_language())`

Reply:
Thanks for the info, your use pattern seems to differ to mine...

I have updated the pull request accordingly.

Reply:
Is there any news on reviewing this change?

Reply:
I just tested this, and the issue looks to be fixed. Sorry that it took a while, I don't use windows much.